js/ui/iconGrid.js

```diff
@@ -1328,6 +1328,18 @@ const PaginatedIconGrid = new Lang.Class({
                                    })
                              });
         }
+    },
+
+    animateShuffling: function(changedItems, removedItems, originalItemData, callback) {
+        // If we try to animate the shuffling before the onComplete callbacks in
+        // closeExtraSpace are called, they are cancelled and we never get to clean
+        // this._extraSpaceData. So, force the cleanup here.
+        if (this._extraSpaceData) {
+            this._extraSpaceData = null;
+            this.emit('space-closed');
```

Comment:
What about adding an `onError` callback to the tween and doing this from there? And if that works, maybe refactoring this code into a helper function, since it's pretty much identical to https://github.com/endlessm/gnome-shell/pull/184/commits/9b6c7272b0157d6fbdcb5984781fe542ff68769a#diff-477f992ae3cb7507f1459b10d86d0e73R1326

Reply:
Oh! Forget it... the actual problem is that the popup's state is not being updated, so this is not correct: we should make sure that its state changes to "closed", and then this hack won't longer be needed

Reply:
I finally found why the onComplete callback was never being called when dragging an icon out of a folder, leaving `this._extraSpaceData` set (messing up the calculations) and the signal not being emitted:

Since such a DnD operation means a change of the GSetting holding the icon grid, what's happening is that a call to `_redisplay()` is happening that causes the icons in the grid that are shifted (because of opening a folder) to be destroyed, which means the in-progress Tweens over them defined in `closeExtraSpace()` will be removed before the transition actually finishes, leaving the onComplete() callback uncalled... therefore the "global" variable `_extraSpaceData` set as it was in the first place.

So, this "trick" works because extraSpaceData is actually never used until you open a folder again, at which point it would be overriden, but it's semantically wrong since you'd be doing that reset + emitting a "space-closed" signal at the time you'd opening a new folder and not before.

I think a better way to do it, even if not 100% correct either, is to always reset `extraSpaceData` after the for loop that creates the tweens in `closeExtraSpace()`, since that what we ensure that it gets always reset regardless of actors being destroyed, and that should not be a problem since `extraSpaceData` is only used from `_calculateDndRow()`, called from `canDropAt()`.